iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -773,8 +796,22 @@ public List<FileStatus> getOutputFiles(List<JobContext> jobContexts) throws IOEx
                 FilesForCommit results = collectResults(numTasks, fileExecutor, table.location(), jobContext,
                         table.io(), false);
                 for (DataFile dataFile : results.dataFiles()) {
-                  FileStatus fileStatus = fileSystem.getFileStatus(new Path(dataFile.path().toString()));
-                  dataFiles.add(fileStatus);
+                  Path filePath = new Path(dataFile.path().toString());
+                  FileStatus fileStatus = fileSystem.getFileStatus(filePath);
+                  parentDirToDataFile.merge(
+                      filePath.getParent(), Lists.newArrayList(fileStatus), (oldList, newList) -> {
+                      oldList.addAll(newList);
+                      return oldList;
+                    });
+                }
+                for (DeleteFile deleteFile : results.deleteFiles()) {
+                  Path filePath = new Path(deleteFile.path().toString());
+                  FileStatus fileStatus = fileSystem.getFileStatus(filePath);
+                  parentDirToDeleteFile.merge(
```

Comment:
same here

Reply:
This is to check the list of files which are part of the same parent directory. Later this is used to decide whether they are eligible candidates for merge.

Reply:
see above comment

Reply:
Done.